src/environment/stella_environment.cpp

```diff
@@ -195,6 +203,26 @@ bool StellaEnvironment::isTerminal() const {
      m_state.getEpisodeFrameNumber() >= m_max_num_frames_per_episode));
 }
 
+void StellaEnvironment::pressSelect(size_t num_steps) {
+  Event* event = m_osystem->event();
+  m_state.pressSelect(event);
+  for (size_t t = 0; t < num_steps; t++) {
+    m_osystem->console().mediaSource().update();
+  }
+  processScreen();
+  processRAM();
+  emulate(PLAYER_A_NOOP, PLAYER_B_NOOP);
```

Comment:
Comparing to softReset(), are we missing a call to `incrementFrame()` here?

Reply:
Honestly, I'm not sure. Do you have any idea how to find that out? The question is in fact whether a frame is necessarily updated once we change the difficulty. I googled it but I couldn't find anything. I am just following @alcinos implementation. What do you want me to do?

Reply:
The frame number inside ALEState is used to determine maximum play length. Should the select button count towards that? Probably, because some games (although we don't support them currently) use select as an action.

Reply:
I added `m_state.incrementFrame();`to the end of this function.